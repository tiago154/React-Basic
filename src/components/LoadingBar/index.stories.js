import React from 'react'
import LoadingBar from '.'
import { withKnobs, color, text } from '@storybook/addon-knobs'

export default {
  title: 'LoadingBar',
  decorators: [withKnobs]
}

export const loadingBar = () => (
  <LoadingBar
    backgroundColor={color('backgroundColor', '#F2C94C')}
    text={text('text', 'Loading...')}
  />
)
