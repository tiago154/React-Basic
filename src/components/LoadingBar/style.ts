import styled from 'styled-components'

interface IPropsStyledContainer {
  backgroundColor: string
}

export const Container = styled.div<IPropsStyledContainer>`
  display: flex;
  justify-content: center;
  align-items: center;
  background: ${({ backgroundColor }) => `${backgroundColor}`};
  border-radius: 15px;
  height: 100px;
  width: 92%;
`

export const Text = styled.p`
  color: #000;
  font-family: 'Fascinate Inline', cursive;
  font-size: 64px;
  font-weight: 900;
  font-style: normal;
`
