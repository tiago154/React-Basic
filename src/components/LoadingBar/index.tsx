import React from 'react'
import { Container, Text } from './style'

interface IProps {
  backgroundColor: string
  text: string
}

const LoadingBar: React.FC<IProps> = ({ backgroundColor, text }) => (
  <Container backgroundColor={backgroundColor}>
    <Text>{text}</Text>
  </Container>
)

export default LoadingBar
