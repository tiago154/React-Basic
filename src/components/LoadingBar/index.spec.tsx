import React from 'react'
import LoadingBar from '.'
import renderer from 'react-test-renderer'

describe('PokemonType', () => {
  it('Renders correctly', () => {
    const tree = renderer
      .create(
        <LoadingBar
          backgroundColor={'#F2C94C'}
          text={'Loading...'}
        />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
