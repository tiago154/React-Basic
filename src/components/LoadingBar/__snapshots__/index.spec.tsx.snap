// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonType Renders correctly 1`] = `
<div
  className="sc-bdVaJa lnUZQr"
>
  <p
    className="sc-bwzfXH akDxl"
  >
    Loading...
  </p>
</div>
`;
